src/julia/02_moelf.jl

```diff
@@ -0,0 +1,31 @@
+const INPUT_PATH = ARGS[1]
+const ALL_LINES = readlines(INPUT_PATH)
+
+function color_dict(a::RegexMatch)
+    num, color = a.captures
+    Dict(color => parse(Int, num))
+end
+
+function line_max(line)
+    RE = r"(\d+)\s(blue|green|red)"
+    parts = eachmatch(RE, line)
+    mapreduce(color_dict, mergewith(max), parts)
+end
+
+# Part I
+function part1_pred(line)
+    dict = line_max(line)
+    return dict["red"] <= 12 &&
+    dict["green"] <= 13 &&
+    dict["blue"] <= 14
+end
+
+p1 = sum(enumerate(ALL_LINES)) do (i, line)
+    return part1_pred(line) ? i : 0
+end
+println(p1)
+
+# Part II
+part2(line) = line |> line_max |> values |> prod
```

Comment:
feels good when Part 2 is free because you guessed the right abstraction based on Part 1